core/modules/savers/git-service.js

```diff
@@ -17,26 +17,23 @@ var base64utf8 = require("$:/core/modules/utils/base64-utf8/base64-utf8.module.j
 /*
 Select the appropriate saver module and set it up
 */
-var GitHubSaver = function(wiki) {
+var GitServiceSaver = function(wiki) {
 	this.wiki = wiki;
 };
 
-GitHubSaver.prototype.save = function(text,method,callback) {
+GitServiceSaver.prototype.save = function(text,method,callback) {
 	var self = this,
-		username = this.wiki.getTiddlerText("$:/GitHub/Username"),
-		password = $tw.utils.getPassword("github"),
-		repo = this.wiki.getTiddlerText("$:/GitHub/Repo"),
-		path = this.wiki.getTiddlerText("$:/GitHub/Path"),
-		filename = this.wiki.getTiddlerText("$:/GitHub/Filename"),
-		branch = this.wiki.getTiddlerText("$:/GitHub/Branch") || "master",
-		endpoint = this.wiki.getTiddlerText("$:/GitHub/ServerURL") || "https://api.github.com",
-		headers = {
-			"Accept": "application/vnd.github.v3+json",
-			"Content-Type": "application/json;charset=UTF-8",
-			"Authorization": "Basic " + window.btoa(username + ":" + password)
-		};
+		titleOfSelectedService = this.wiki.getTiddlerText("$:/GitService/ServiceType"),
+		service = require(titleOfSelectedService),
+		username = this.wiki.getTiddlerText("$:/GitService/Username"),
+		password = $tw.utils.getPassword("gitservice"),
+		repo = this.wiki.getTiddlerText("$:/GitService/Repo"),
+		path = this.wiki.getTiddlerText("$:/GitService/Path"),
+		filename = this.wiki.getTiddlerText("$:/GitService/Filename"),
+		branch = this.wiki.getTiddlerText("$:/GitService/Branch") || "master",
+		apiUrl = this.wiki.getTiddlerText("$:/GitService/ServerURL") || this.wiki.getTiddler(titleOfSelectedService).fields['default-api-url'];
```

Comment:
Here you're using a field of a JS module to hold the default API url. It's clearer to keep everything in JavaScript, and have the JS module export the value.

Reply:
@Jermolene Why so many wiki.getTiddlerText? Shouldn't those fields merged into a single tiddler, and use different fields of a tiddler?

And what do you mean by "keep everything in JavaScript"?

Reply:
Oh, I know!

The text you added to `core/language/en-GB/ControlPanel.multids` becomes several tiddlers on boot! And they are automatically registered as control panel fields, so you are actually retrieving control panel fields!

Reply:
>Why so many wiki.getTiddlerText? Shouldn't those fields merged into a single tiddler, and use different fields of a tiddler?

Experience shows, that separating config values has advantages. eg: config values are also be used by the UI templates. If you change 1 tiddler, the refresh mechanism only updates those parts that use this tiddler. ...

Refresh is triggered by tiddler change and not per field. So if you change 1 of 100 fields, you'd need to update them all

Reply:
@Jermolene I wanted to minimize the code duplication, so I wanted to get the value from one place, either I'm using it in Javascript or in a tiddler. What is the right solution for this?

Reply:
@Jermolene 

> have the JS module export the value

I don't know how to fix it. For a widget, `setVariable()` works, but this is a saver module where the method is unavailable (as I know). How do I create a "global" variable? Or is it so minimal duplication not to deal with it and declare the URL in two places (in the config interface and in Javascript)?